text/0025-core-compute-resources-as-plugins.md

```diff
@@ -0,0 +1,77 @@
+- Feature Name: Core Compute Resources as Plugins
+- Start Date: 2016-10-21
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Core Compute Resources should move to plugins
+
+# Motivation
+[motivation]: #motivation
+
+Core has been set up to allow compute resources as plugins for some while, and
+we already have several examples of this in use (Xen, Azure, DigitalOcean,
+etc). This has brought benefits:
+
+* able to release independatly of Core
```

Comment:
Fog is already breaking down into multiple providers anyway (fog-openstack, fog-google, fog-azure, etc) and I'm sure they'd welcome PRs to continue that work.

So, my assumption is that we can ship a version of Fog-core in foreman-compute (which *should* prevent incompatibility issues), and plugins can depend on the appropriate Fog subpackage. I agree that it's not perfect, but it feels like it grants CRs more flexibility than we currently have.